oci/spec_opts.go

```diff
@@ -1262,15 +1262,15 @@ var ErrNoShmMount = errors.New("no /dev/shm mount specified")
 // The size value is specified in kb, kilobytes.
 func WithDevShmSize(kb int64) SpecOpts {
 	return func(ctx context.Context, _ Client, c *containers.Container, s *Spec) error {
-		for _, m := range s.Mounts {
-			if m.Source == "shm" && m.Type == "tmpfs" {
-				for i, o := range m.Options {
-					if strings.HasPrefix(o, "size=") {
-						m.Options[i] = fmt.Sprintf("size=%dk", kb)
-						return nil
+		for mi, m := range s.Mounts {
+			if m.Destination == "/dev/shm" && m.Source == "shm" && m.Type == "tmpfs" {
```

Comment:
why limit at /dev/shm?

Reply:
I don't think it should affect all mounts with source `shm` and type `tmpfs`, because the user may also specify such a mount, and the name of the function and the comments indicate that it is used to set the /dev/shm

Reply:
Hmm. Looks like you want to change the behavior. I don't think we should change it if you don't know the impact scope.

Reply:
This function is not used anywhere else but in tests

In the original function, only the first mount is handled, do you feel it is reasonable?

Reply:
I searched containerd/moby/rancher org and looks like no one use it. SGTM.